### PR TITLE
Feature/ternary expression js compiler

### PR DESCRIPTION
**Pull request by TobiasWienand**
Addresses issue #437 

This PR enables compilation of ternary expressions (i.e. a ? b : c). Known as conditional expression in the babel js parser

### CODE REVIEW

File: Sources/Fuzzilli/Compiler/Parser/parser.js [hunk 1]
```diff
@@ -519,6 +519,15 @@ function parse(script, proto) {
                 let argument = visitExpression(node.argument);
                 return makeExpression('UnaryExpression', { operator, argument });
             }
+            case 'ConditionalExpression': {
+
```
Comment (saelo) [inline Sources/Fuzzilli/Compiler/Parser/parser.js:523]: Nit: maybe remove this blank line

File: Sources/Fuzzilli/Compiler/Parser/parser.js [hunk 2]
```diff
@@ -571,8 +580,8 @@ protobuf.load(astProtobufDefinitionPath, function(err, root) {
 
     // Uncomment this to print the AST to stdout (will be very verbose).
     //console.log(JSON.stringify(ast, null, 2));
-
-    const AST = root.lookupType('compiler.protobuf.AST');
+    
+const AST = root.lookupType('compiler.protobuf.AST');
```
Comment (saelo) [inline Sources/Fuzzilli/Compiler/Parser/parser.js:584]: Accidental change (indention)?

File: Tests/FuzzilliTests/CompilerTests/basic_expressions.js [hunk 3]
```diff
@@ -26,6 +26,7 @@ output(a >> b);
 output(a >>> b);
 output(a && b);
 output(a || b);
+output(a > b ? a : b);
```
Comment (saelo) [inline Tests/FuzzilliTests/CompilerTests/basic_expressions.js:29]: Maybe also add the opposite(i.e. `a < b ? a : b`) to test both directions

File: Sources/Fuzzilli/Compiler/Compiler.swift [hunk 4]
```diff
@@ -502,9 +502,16 @@ public class JavaScriptCompiler {
         guard let expr = node.expression else {
             throw CompilerError.invalidASTError("missing concrete expression in expression node")
         }
-
+
```
Comment (saelo) [inline Sources/Fuzzilli/Compiler/Compiler.swift:505]: Nit: I think you introduced some whitespace here and in parser.js, maybe remove that again

**Review by saelo — COMMENTED** [review]
Really nice, thanks!

File: Sources/Fuzzilli/Compiler/Parser/parser.js (hunk 1 above)
Comment (TobiasWienand) [reply Sources/Fuzzilli/Compiler/Parser/parser.js]: Here I also removed two more blank lines such that this case statements style aligns with the rest of the code. The other comments below this one have also been addressed with commit 3b71e9fab14ae4b48bf87b3817064dfe823e1091

**Comment by saelo** [comment]
Thanks! :)

**Pull request merged**